app/api/resources/user.py

```diff
@@ -498,3 +498,111 @@ def get(cls):
             return messages.USER_NOT_FOUND, HTTPStatus.NOT_FOUND
 
         return dashboard, HTTPStatus.OK
+
+@users_ns.route("user/forgot_password")
+class ForgotPassword(Resource):
+    @classmethod
+    @users_ns.doc(f"forgot password")
+    @users_ns.response(HTTPStatus.OK, f"{messages.PASSWORD_RESET_MAIL_MESSAGE}")
+    @users_ns.response(HTTPStatus.BAD_REQUEST, f"{messages.EMAIL_FIELD_IS_MISSING}")
+    @users_ns.response(HTTPStatus.FORBIDDEN, f"{messages.USER_HAS_NOT_VERIFIED_EMAIL_BEFORE_LOGIN}")
+    @users_ns.response(HTTPStatus.NOT_FOUND, f"{messages.USER_IS_NOT_REGISTERED_IN_THE_SYSTEM}")
+    @users_ns.expect(forgot_password_change_request_data_model, validate=True)
+ 
+    def post(self):
+        """ Updates User's password by sending reset link to email
+ 
+        Args: 
+          email : Verified email of the user
+ 
+        Return:
+          token: a token link/reset link will be shared on verified mail of user
+        """
+ 
+        data = request.json
+ 
+        if not data:
+            return messages.EMAIL_FIELD_IS_MISSING, HTTPStatus.BAD_REQUEST
+ 
+        is_valid = validate_forgot_password_email(data)
+ 
+        if is_valid!={}:
+            return is_valid, HTTPStatus.BAD_REQUEST
+        
+        user = DAO.get_user_by_email(data["email"])
+ 
+        if user is None:
+            return messages.USER_IS_NOT_REGISTERED_IN_THE_SYSTEM, HTTPStatus.NOT_FOUND
+ 
+        if not user.is_email_verified:
+            return messages.USER_HAS_NOT_VERIFIED_EMAIL_BEFORE_LOGIN, HTTPStatus.FORBIDDEN    
+ 
+        send_email_reset_password_message(user.name, data["email"])
+ 
+        return messages.PASSWORD_RESET_MAIL_MESSAGE, HTTPStatus.OK
+ 
+@users_ns.route("user/reset_password")
+@users_ns.response(HTTPStatus.OK, f"{messages.PASSWORD_SUCCESSFULLY_UPDATED}")
+@users_ns.response(
+    HTTPStatus.UNAUTHORIZED,
+    "%s\n%s\n%s"
+    % (
+        messages.TOKEN_HAS_EXPIRED,
+        messages.TOKEN_IS_INVALID,
```

Comment:
Is this message used anywhere?

Reply:
are you talking about that "Token Is Invalid" message? I don't remember exactly where I have to search for it ! will inform you about it soon :)

Reply:
@robotjellyzone @Tlazypanda I tried to test this with an invalid password but the message I get is expired password.
![Screenshot from 2020-05-03 12-28-04](https://user-images.githubusercontent.com/26095715/80919752-c5d5d900-8d39-11ea-8f11-5ce8bec40280.png)

Reply:
Oh!! i have to work on these responses as well, sorry @rpattath i will correct it soon!! its very tiring to handle such a big code in one go . I have to search where this "Token Is Invalid" message goes to. That's a great testing Roshni !! you are becoming a PRO in it 👍 🥇 . Thank you for testing this as well :) will correct it soon!!

Reply:
Hi @rpattath can you please tell me which token were you using due to which you got expired message ? i am testing it now . sorry got busy :)

Reply:
Hi @robotjellyzone, I  removed a few characters from a valid token and was using that.

Reply:
Ok , i will try this one to confirm it again:)